### DISCUSSION TITLE
OTX integration script

### DISCUSSION THREAD

**Discussion by greatapoc** [Security-Onion-Solutions/securityonion#1876] [Unsupported Versions] [answered]
I noticed the documentation for the OTX integration had been updated under issue #1678 so I decided to give it a go. Just had some minor things that need to be updated. 

The initial script download needs to have chmod to make it executable. Likewise at the end /usr/sbin/zeek_otx.py was not executable on my system and needed to be changed. That script also fails to run with the following error:
[root@securityonion sbin]# zeek_otx.py
Traceback (most recent call last):
  File "/sbin/zeek_otx.py", line 9, in <module>
    from configparser import ConfigParser
ImportError: No module named configparser

EDIT: Scratch that, I see now it needs to be run with python3 and it succeeds.

/nsm/zeek/logs/current/intel.log does not exist on my machine.

**Comment by weslambert** [comment]
Hi @greatapoc ,

What OS did you install this on?

I haven't had to `chmod +x` in my testing, but I can make it explicit.

To be clear, did it not invoke Python3 by itself when running the script?  I assumed it would be linked to Py3 by default.

Thanks,
Wes

**Reply by greatapoc** [reply]
Hi Wes,

CentOS 7 from the SO ISO. The cron job specifies python3 in the call so that runs fine.

**Reply by weslambert** [reply]
Okay, I will have to test it.

**Comment by tommorgan365** [comment]
Wes,
Just to clarify - do you need to run this on each individual sensor or at the manager?

**Reply by weslambert** [reply]
This is only needed on the manager.  The config should propagate to the forward nodes.

**Reply by polivia** [reply]
I don't think the propagation to the sensor nodes is working.  I see the otx.dat in /opt/so/conf/zeek/policy/custom/zeek-otx/ on the manager node, but the sensor node is not getting a copy.

**Reply by weslambert** [reply]
Okay, will have to test to ensure this is happening as it should.

**Reply by tommorgan365** [reply]
Checking back in on this one Wes.

**Reply by weslambert** [reply]
I just committed a fix here: https://github.com/weslambert/securityonion-otx/commit/88329d074e1b67dd5f441e8d7933fb43bd1b63ff.

Please note that the directory on the manager should be `/opt/so/saltstack/local/salt/zeek/policy/custom/zeek-otx`, as opposed to the previous `/opt/so/conf/...` directory.  This should allow the config to be replicated to downstream nodes.

**Reply by polivia** [reply]
Your most recent changes on the to the GIT repo fixed things.  I can confirm it's now pushing the files to the sensor nodes.

**Comment by weslambert** [comment, marked as answer]
This has been fixed in the repo.
